src/components/Editable/useOnPaste.ts

```diff
@@ -43,86 +36,26 @@ const useOnPaste = ({
         e.preventDefault()
         return
       }
-
-      // pasting from mobile copy (e.g. Choose "Share" in Twitter and select "Copy") results in blank plainText and htmlText
-      // the text will still be pasted if we do not preventDefault, it just won't get stripped of html properly
-      // See: https://github.com/cybersemics/em/issues/286
       if (plainText || htmlText) {
         e.preventDefault()
 
-        // import into the live thoughts
-        // neither ref.current is set here nor can newValue be stored from onChange
-        // not sure exactly why, but it appears that the DOM node has been removed before the paste handler is called
-        const { cursor, cursorOffset: cursorOffsetState } = store.getState()
-        const path = cursor && equalPath(cursor, simplePath) ? cursor : simplePath
-
-        // If transient first add new thought and then import the text
-        if (transient) {
-          dispatch(
-            newThought({
-              at: rootedParentOf(store.getState(), path),
-              value: '',
-            }),
-          )
-        }
-
-        const text = htmlText
-          ? strip(htmlText, { preserveFormatting: emText, stripColors: !emText }).replace(/\n\s*\n+/g, '\n') // Clean HTML from clipboard
-          : escapeHtml(plainText.trim()) // Escape plain text from clipboard
-
-        // Is this an adequate check if the thought is multiline, or do we need to use textToHtml like in importText?
-        const multiline = plainText.trim().includes('\n')
-
-        // Check if the text is markdown, if so, prefer importText over importFiles
-        const markdown = isMarkdown(text)
-        // Resumable imports (via importFiles) import thoughts one at a time and can be resumed if the page is refreshed or there is another interruption. They have a progress bar and they allow duplicates pending descendants to be loaded and merged.
-        // Non-resumable imports (via importText), in contrast, are atomic, fast, and preserve the browser selection. Due to the lack of support for duplicates pending descendants, they are only used for single line imports.
-        if (!multiline || markdown) {
-          dispatch(
-            importText({
-              // use caret position to correctly track the last navigated point for caret
-              // calculated on the basis of node type we are currently focused on. `state.cursorOffset` doesn't really keep track of updated caret position when navigating within single thought. Hence selection.offset() is also used depending upon which node type we are on.
-              caretPosition: (selection.isText() ? selection.offset() || 0 : cursorOffsetState) || 0,
```

Comment:
It looks like some of the inline comments have been lost in this refactor. Could you go back through and make sure that all comments are preserved?